evaluator/functions.go

```diff
@@ -1319,3 +1346,57 @@ func sleepFn(tok token.Token, args ...object.Object) object.Object {
 
 	return NULL
 }
+
+// source("fileName")
+// aka src(), include(), import(), require()
+var sourceMaxRecursionDepth = 5
+
+func sourceFn(tok token.Token, args ...object.Object) object.Object {
+	// limit source file recursion depth
+	if sourceMaxRecursionDepth <= 0 {
+		return newError(tok, "maximum source file recursion depth exceeded")
+	}
+	// mark this source recursion level
+	sourceMaxRecursionDepth--
+
+	err := validateArgs(tok, "source", args, 1, [][]string{{object.STRING_OBJ}})
+	if err != nil {
+		return err
+	}
+	// load the source file
+	fileName, _ := util.ExpandPath(args[0].Inspect())
+	code, error := ioutil.ReadFile(fileName)
+	if error != nil {
+		// source file path does not exist
+		return newError(tok, "source file not found: %s", fileName)
```

Comment:
```suggestion
		return newError(tok, "error when reading %s: %s", fileName, err.Error())
```

It could be that the problem is that the file is not readable, so I wouldn't just throw a generic error saying the file is not found.

Reply:
Done